woocommerce/woocommerce-ios
# Reusable components for product discounts view

Closes: #10915

## Description:
This PR addresses feedback regarding the product discounts view logic, and simplifies it to render the different elements by extracting these into separate, reusable components, as well as leaving the main View as a simple switch case rather than relying in multiple `.renderedIf()` statements. So now is clearer what are we rendering, and when.

This PR does not change any logic, neither what the views are rendering, just changes how we compose them. Product discounts still work and look as before the changes.

## Changes:
- Extracted the main view into a `LineDetailView` reusable component, which stacks two subviews vertically, then any components passed to the 2nd view horizontally.
- Collapsed the input fields for fixed and percentage discounts into an unique `InputField` component.
- Collapsed the buttons to toggle between fixed and percentage discounts into the computed View `discountTypeButtonToggle`

## Testing instructions
* Go to Orders > `+` > `+ Add Products` > Select a product > Toggle the chevron to expand additional options > Tap `+ Add discount`
* Observe that: 
  * Switching between `$` and `%` changes the input details (Fixed price discount vs Percentage discount)
  * Entering a value in `$` or `%` input fields makes the discount appear below and calculates the discount on a fixed or percentage manner.

Please note there are [known bugs](https://github.com/woocommerce/woocommerce-ios/issues/10863) to the discounts implementation (like negative discounts, or discounting more than the value of the item), this PR does not address those, but just refactors the views.

| Fixed | Percentage |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 - 2023-10-17 at 06 41 29](https://github.com/woocommerce/woocommerce-ios/assets/3812076/eee01222-03fa-4b78-b98d-783d0fbde0c2) | ![Simulator Screenshot - iPhone 15 - 2023-10-17 at 06 41 37](https://github.com/woocommerce/woocommerce-ios/assets/3812076/52fb4d4d-82c0-4d47-b716-8510de7dfdbe) |
